app/services/page_creator.rb

```diff
@@ -73,33 +67,30 @@ def multilingual_campaign_uri
     end
 
     def page_type
-      raise "Not Implemented"
+      self.class.name.demodulize.underscore
```

Comment:
Hopefully not but: is there a chance this might break on an other types of pages? e.g. non donation or petitions. Or do we only process donation / petition pages?

Reply:
This class is only used for Donation and Petition pages right now. There's also e.g. the PageFollowUpCreator which creates the follow up resource. This'll be easy to override in classes that would inherit from Base the same way as Petition and Donation used to - I just find that this default implementation would catch most class names so I didn't need to override this method for Donation and Petition separately.